croissant-rdf/pyproject.toml

```diff
@@ -0,0 +1,119 @@
+[project]
+name = "croissant-rdf"
+version = "0.1.2"
+description = "Tools for working with RDF from Croissant JSON-LD resources."
+readme = "README.md"
+license = { file = "LICENSE" }
+authors = [{ name = "David Steinberg", email = "david@resium.com" }]
+maintainers = [{ name = "David Steinberg", email = "david@resium.com" }]
+keywords = [
+    "CroissantML",
+    "RDF",
+    "JSON-LD",
+    "SPARQL",
+]
+classifiers = [
+    "License :: OSI Approved :: MIT License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python :: 3",
+]
+
+requires-python = ">=3.8"
+dependencies = [
+    "rdflib >=6.0.0",
+    "huggingface_hub >=0.29.1",
```

Comment:
That could be relevant to keep these dependencies optional. Probably openml users won't need kaggle, etc.

We have a mechanism in place in mlcroissant to do this.

Reply:
I understand the impetus of separating out dependencies. I am partial to including them because it adds complexity that is passed onto the user and these dependencies are pretty lightweight (basically wrappers for requests). Basically the benefit of having access to all the Croissant registries out of the box outweighs the negative of having a slightly larger installation.

Reply:
> these dependencies are pretty lightweight (basically wrappers for requests)

SGTM